arelle/ModelInstanceObject.py

```diff
@@ -941,6 +944,17 @@ def startDatetime(self, value):
             elt.text = XmlUtil.dateunionValue(value)
             xmlValidate(self.modelXbrl, elt)
 
+    @property
+    def endDate(self) -> date:
+        """
+        :return: endDate or instant attribute as date, *not* adjusted by a day for midnight values
+        """
+        try:
+            return self._endDate
+        except AttributeError:
+            self._endDate = XmlUtil.datetimeValue(XmlUtil.child(self.period, XbrlConst.xbrli, ("endDate", "instant"))).date()
```

Comment:
Does this return `2022-12-31` for `2023-01-01T00:00:00`?  I believe it should, so I think start with `endDatetime` but subtract one day if (hour,minute,second,microsecond)==(0,0,0,0).  Similarly for `instantDate`.